gap_statistic/optimalK.py

```diff
@@ -127,14 +135,107 @@ def __call__(
                     "n_clusters": gap_calc_result.n_clusters,
                     "gap_value": gap_calc_result.gap_value,
                     "ref_dispersion_std": gap_calc_result.ref_dispersion_std,
+                    "sdk": gap_calc_result.sdk,
+                    "sk": gap_calc_result.sk,
+                    "gap*": gap_calc_result.gap_star,
+                    "sk*": gap_calc_result.sk_star,
                 },
                 ignore_index=True,
             )
+            gap_df['gap_k+1'] = gap_df['gap_value'].shift(-1)
+            gap_df['gap*_k+1'] = gap_df['gap*'].shift(-1)
+            gap_df['sk+1'] = gap_df['sk'].shift(-1)
+            gap_df['sk*+1'] = gap_df['sk*'].shift(-1)
+            gap_df['diff'] = gap_df['gap_value'] - gap_df['gap_k+1'] + gap_df['sk+1']
+            gap_df['diff*'] = gap_df['gap*'] - gap_df['gap*_k+1'] + gap_df['sk*+1']
+
+        # drop auxilariy columns
+        gap_df.drop(
+            labels=['sdk', 'sk', 'sk*', 'gap_k+1', 'gap*_k+1', 'sk+1',
+                    'sk*+1'],
```

Comment:
Did `black` put this on a new line? If not, could you move it so the same line?

Reply:
I disabled my auto-black on save for your messy project. :)

Reply:
Done.